docs/introduction/03-getting-started.md

```diff
@@ -98,18 +98,161 @@ them for a rapid feedback cycle.
+  @Prop() middle: string;
+  @Prop() last: string;
+```
+`@Prop()` tells Stencil that the property is public to the component, and allows Stencil to update what's rendered when any of these public properties change.
+We'll see how this works after discussing the `render()` function.
+
+In order for the component to render something to the screen, we must declare a [`render()` function](../components/templating-and-jsx.md#basics) that returns JSX.
+If you're not sure what JSX is, be sure to reference the [Using JSX](../components/templating-and-jsx.md) docs.
+
+The quick idea is that our render function needs to return a representation of the HTML we want to push to the DOM.
+
+```tsx
+  private getText(): string {
+    return format(this.first, this.middle, this.last);
+  }
+  
+  render() {
+    return <div>Hello, World! I'm {this.getText()}</div>;
+  }
+```
+
+This component's `render()` returns a `<div>` element, containing text to render to the screen.
+
+The `render()` function uses all three class members decorated with `@Prop()`, through the `getText` function.
+Declaring private functions like `getText` helps pull logic out of the `render()` function's JSX.
+
+Any property decorated with `@Prop()` is also automatically watched for changes.
+If a user of our component were to change the element's `first`, `middle`, or `last` properties, our component would fire its `render()` function again, updating the displayed content.
+
+## Component Generator
+
+The `create-stencil` CLI can generate new components for you.
+If you used one of the starters, you can simply run the `generate` npm script in your project, which will start the interactive generator.
+
+```shell npm2yarn
+npm run generate
+```
+
+Or, you can invoke the Stencil CLI directly with the `generate` command (`g` for short).
+If you don't have `stencil` installed globally, prefix the command with `npx`.
+
+```shell
+stencil generate
+```
+
+You can optionally pass the component tag name directly to the command.
+The component tag name needs to be lowercase and contain at least one dash ('-').
+
+The generator will ask you which files to generate. 
+This allows you to bootstrap a stylesheet as well as spec and e2e tests along with the component file.
+
+All components will be generated within the `src/components` folder.
+Within that directory, a folder will be created with the same name as the component tag name you provided, and within that folder the files will be generated.
+It is also possible to specify one or multiple sub-folders to generate the component in.
+
+For example, if you specify `pages/page-home` as the component tag name, the files will be generated in `src/components/pages/page-home`.
+
+```shell
+stencil generate pages/page-home
+```
+
+```plain
+src
+ |- components
+     |- pages
+         |- page-home
+             |- page-home.css
+             |- page-home.e2e.ts
+             |- page-home.spec.ts
+             |- page-home.tsx
+```
```

Comment:
Made a dummy directory and then ran `tree` on it to get slightly nicer symbols:

````suggestion
```plain
src
└── components
    └── pages
        └── page-home
            ├── page-home.css
            ├── page-home.e2e.ts
            ├── page-home.spec.ts
            └── page-home.tsx
```
````

looks like this rendered in Docusaurus:

<img width="811" alt="Screenshot 2023-03-02 at 10 30 10 AM" src="https://user-images.githubusercontent.com/6207644/222473418-9969ed8b-7dea-406b-9a99-32774a8b4dbb.png">